### PR TITLE
[NT-940] Activate Creator Details Experiment

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Activates the `project_page_conversion_creator_details` experiment.

# 🤔 Why

A continuation of the work in #1086, this conditionally activates the experiment if the fetching of the project creator details succeeds.

# 🛠 How

- Added the experiment activation.
- Added logic to only activate the experiment if we were able to successfully retrieve the project creator details from GraphQL as this is a secondary network request.

# ✅ Acceptance criteria

Navigate to a project page.

- [ ] If `fetchProjectCreatorDetails` **fails** and you are in `variant1` the VM should output `(nil, true)` for `ProjectCreatorDetailsData`.
- [x] If `fetchProjectCreatorDetails` **succeeds** and you are in `variant1` the VM should output `(result, true)` for `ProjectCreatorDetailsData`.
- [x] If you are in the `control` group, the VM should output `(nil, true)` for `ProjectCreatorDetailsData`.

### CODE REVIEW

File: Library/OptimizelyExperiment.swift [hunk 1]
```diff
@@ -5,6 +5,7 @@ public enum OptimizelyExperiment {
   public enum Key: String {
     case pledgeCTACopy = "pledge_cta_copy"
     case nativeProjectPageCampaignDetails = "native_project_page_campaign_details"
+    case projectPageConversionCreatorDetails = "project_page_conversion_creator_details"
```
Comment (cdolm92) [inline Library/OptimizelyExperiment.swift]: Looks like the name of the experiment is `native_project_page_conversion_creator_details`

File: Library/ViewModels/ProjectPamphletContentViewModelTests.swift [hunk 2]
```diff
@@ -398,14 +398,19 @@ final class ProjectPamphletContentViewModelTests: TestCase {
...
     let project = Project.template
 
     let envelope = ProjectCreatorDetailsEnvelope(backingsCount: 25, id: "123", launchedProjectsCount: 50)
 
     let mockService = MockService(fetchProjectCreatorDetailsResult: .success(envelope))
 
-    withEnvironment(apiService: mockService) {
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeProjectPageConversionCreatorDetails.rawValue: OptimizelyExperiment.Variant.variant1.rawValue
```
Comment (nativeksr) [inline Library/ViewModels/ProjectPamphletContentViewModelTests.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/kickstarter/ios-oss/blob/8676c26a01ecffa0f982d0e5b09f2191cd795621/Library/ViewModels/ProjectPamphletContentViewModelTests.swift#L410"></span>Line should be 110 characters or less: currently 131 characters<br />
<code>line_length</code> <code>ProjectPamphletContentViewModelTests.swift:410</code></td>
    </tr>
  </tbody>
</table>

File: Library/ViewModels/ProjectPamphletContentViewModelTests.swift [hunk 3]
```diff
@@ -471,6 +476,84 @@ final class ProjectPamphletContentViewModelTests: TestCase {
...
+  func testLoadProjectIntoDataSource_CreatorDetailsLoaded_ControlVariant() {
+    let project = Project.template
+
+    let envelope = ProjectCreatorDetailsEnvelope(backingsCount: 25, id: "123", launchedProjectsCount: 50)
+
+    let mockService = MockService(fetchProjectCreatorDetailsResult: .success(envelope))
+
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeProjectPageConversionCreatorDetails.rawValue: OptimizelyExperiment.Variant.control.rawValue
```
Comment (nativeksr) [inline Library/ViewModels/ProjectPamphletContentViewModelTests.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/kickstarter/ios-oss/blob/8676c26a01ecffa0f982d0e5b09f2191cd795621/Library/ViewModels/ProjectPamphletContentViewModelTests.swift#L488"></span>Line should be 110 characters or less: currently 130 characters<br />
<code>line_length</code> <code>ProjectPamphletContentViewModelTests.swift:488</code></td>
    </tr>
  </tbody>
</table>

File: Library/ViewModels/ProjectPamphletContentViewModelTests.swift [hunk 4]
```diff
@@ -539,4 +622,78 @@ final class ProjectPamphletContentViewModelTests: TestCase {
...
   }
+
+  func testLoadProjectIntoDataSource_CreatorDetailsFailure_ExperimentalVariant() {
+    let project = Project.template
+
+    let mockService = MockService(fetchProjectCreatorDetailsResult: .failure(.invalidInput))
+
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeProjectPageConversionCreatorDetails.rawValue: OptimizelyExperiment.Variant.variant1.rawValue
```
Comment (nativeksr) [inline Library/ViewModels/ProjectPamphletContentViewModelTests.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/kickstarter/ios-oss/blob/8676c26a01ecffa0f982d0e5b09f2191cd795621/Library/ViewModels/ProjectPamphletContentViewModelTests.swift#L633"></span>Line should be 110 characters or less: currently 131 characters<br />
<code>line_length</code> <code>ProjectPamphletContentViewModelTests.swift:633</code></td>
    </tr>
  </tbody>
</table>

**Pull request merged**